column_index.go

```diff
@@ -386,12 +416,39 @@ func (i *int64ColumnIndexer) IndexPage(numValues, numNulls int64, min, max Value
 }
 
 func (i *int64ColumnIndexer) ColumnIndex() format.ColumnIndex {
-	return i.columnIndex(
-		splitFixedLenByteArrays(unsafecast.Int64ToBytes(i.minValues), 8),
-		splitFixedLenByteArrays(unsafecast.Int64ToBytes(i.maxValues), 8),
-		orderOfInt64(i.minValues),
-		orderOfInt64(i.maxValues),
-	)
+	if cpu.IsBigEndian {
+		minLen := len(i.minValues)
+		maxLen := len(i.maxValues)
+		byteMin := make([]byte, minLen*8)
+		idx := 0
+		for k := 0; k < minLen; k++ {
+			minVal := (i.minValues)[k]
+			binary.LittleEndian.PutUint64(byteMin[idx:(8+idx)], uint64(minVal))
+			idx += 8
+		}
```

Comment:
We repeat these loops twice in each function, could we factor these out in separate functions that we can reuse?